aten/src/ATen/native/Fill.cpp

```diff
@@ -10,7 +10,39 @@ namespace native {
 
 // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ fill ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+namespace {
+  template <typename scalar_t>
+  inline void fill_fast(Tensor& self, Scalar value_scalar) {
+    auto value = value_scalar.to<scalar_t>();
+    scalar_t * dptr = reinterpret_cast<scalar_t *>(self.data_ptr());
+    *dptr = value;
+  }
+} // namspace
+
 Tensor& fill_out(Tensor& self, Scalar value) {
+  // When filling a number to 1-element tensor, we want to skip
+  // everything but manipulate data ptr directly.
+  // Ideally this fast pass should be implemented in TensorIterator,
+  // but we also want to skip compute_types which in not avoidable
+  // in TensorIterator for now.
+  if (self.numel() == 1) {
+     // FIXME: I only listed a few types here since I'm looking for
+     // an API to map ScalarType to C++ type but haven't found it.
```

Comment:
I think you're looking for a dispatch macro: https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/Dispatch.h#L53